server/rules-engine/index.js

```diff
@@ -0,0 +1,175 @@
+const DEBUG = false
+const Engine = require('json-rules-engine').Engine
+const enums = require('./enums')
+const Rule = require('json-rules-engine').Rule
+const operators = require('./operators')
+const factHandler = require('./factHandler')
+
+const _buildStandardRules = function (conditions, ruleName, eventType) {
+  if (DEBUG) console.log('Setting up standard rules with conditions ', conditions)
+  return conditions.map(condition => new Rule({
+    conditions: {
+      all: [condition]
+    },
+    event: {
+      type: eventType
+    },
+    name: ruleName,
+    priority: 5
+  }))
+}
+
+const _setupStandardRule = function (conditions, ruleName, eventType) {
+  const rules = _buildStandardRules(conditions, ruleName, eventType)
+  this.addRule(rules)
+  this.engine.on('success', function (event, almanac, ruleResult) {
+    if (event.type === eventType) {
+      almanac.addRuntimeFact(enums.ruleRejected, true)
+      if (DEBUG) {
+        almanac.factValue('ref').then(ref => {
+          console.log('Rule passed, rejecting parcel:', ref)
+        })
+      }
+    }
+  })
+}
+
+const _buildAcceptedItemsRules = function () {
+  return [new Rule({
+    conditions: {
+      all: [{
+        fact: enums.ruleRejected,
+        operator: 'notEqual',
+        value: true
+      }]
+    },
+    event: {
+      type: enums.acceptedEventName
+    },
+    priority: 1,
+    name: 'matchAcceptedItems'
+  })]
+}
+
+const _setupAcceptedItemsRule = function (funcToCall) {
+  const rules = _buildAcceptedItemsRules()
+  this.addRule(rules)
+  this.engine.on('success', function (event, almanac, ruleResult) {
+    if (event.type === enums.acceptedEventName) {
+      if (typeof funcToCall !== 'undefined') {
+        funcToCall(event, almanac, ruleResult)
+      }
+    }
+  })
+}
+
+const _buildCalculationRule = function (ruleDef) {
+  return new Rule(ruleDef)
+}
+
+const _setupCalculationRules = function (ruleDefs) {
+  this.addRule(ruleDefs.map(ruleDef => _buildCalculationRule(ruleDef)))
+}
+
+const _setupFullRun = function (rulesJson, dataJson, successFunc) {
+  const enabledRules = this.enabledRules(rulesJson)
+  if (DEBUG) console.log('enabledRules', enabledRules)
+  const conditions = this.conditionsFromRules(enabledRules)
+  if (DEBUG) console.log(conditions)
+  this.setupStandardRule(conditions, 'Standard', 'standard')
+  this.setupAcceptedItemsRule(successFunc)
+  const calculatedFacts = this.factHandler.getCalculatedFacts(enabledRules)
+  const calcRules = this.factHandler.buildCalculationRules(calculatedFacts)
+  this.loadCalculationRules(calcRules)
+}
+
+const _doFullRun = async function (rulesJson, dataJson, passedFacts, successFunc) {
+  const retVal = []
+  this.setupFullRun(rulesJson, dataJson, successFunc)
+  if (DEBUG) {
+    for (const rule of this.engine.rules) {
+      console.log(rule.toJSON())
+    }
+  }
+  dataJson.map(data => {
+    retVal.push(this.runEngine(data, passedFacts))
+  })
```

Comment:
I think this use of `map` and the subsequent `await Promise.all` could be cleaner. Something like:
```
const results = await Promise.all(
  dataJson.map(data => this.runEngine(data, passedFacts))
)
```
(`results` is then an array of the resolved rules engine results)